From 9c1bd48929ad7c6b55d4486e7c519c778f9900d6 Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Mon, 14 Oct 2024 15:27:28 +0800
Subject: [PATCH] mac80211: mtk: set IEEE80211_TX_CTL_USE_MINRATE when probing
 station

The TxS may not be reported to driver correctly when we set BA_DISALBE = 0.

When mac80211 set IEEE80211_TX_CTL_USE_MINRATE, mt76 would use fixed rate
and set BA_DISABLE = 1 to transmit the packet. So mt76 can receive TxS
correctly.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 net/mac80211/cfg.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/net/mac80211/cfg.c b/net/mac80211/cfg.c
index 80fba54..4818dca 100644
--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -3957,7 +3957,8 @@ static int ieee80211_probe_client(struct wiphy *wiphy, struct net_device *dev,
 	info = IEEE80211_SKB_CB(skb);
 
 	info->flags |= IEEE80211_TX_CTL_REQ_TX_STATUS |
-		       IEEE80211_TX_INTFL_NL80211_FRAME_TX;
+		       IEEE80211_TX_INTFL_NL80211_FRAME_TX |
+		       IEEE80211_TX_CTL_USE_MINRATE;
 	info->band = band;
 
 	skb_set_queue_mapping(skb, IEEE80211_AC_VO);
-- 
2.45.2

